src/main/scala/game/Game.scala

```diff
@@ -37,33 +66,22 @@ object Game extends App {
       """
     )
 
-    val userInput: String = scala.io.StdIn.readLine()
-
-    val maybePlayAgain = parsePlayerMove(userInput)
-      .fold(
-        invalidCommad => {
-          println(invalidCommad.getFormattedErrorMsg());
-          Some(PlayAgainCommand);
-        },
-        cmd =>
-          cmd match {
-            case userMove: PlayCommand => {
-              val cpuMove = provideCPUMove()
-              println(s"[User] ${userMove.toString()} <-> ${cpuMove.toString()} [CPU]")
-
-              val matchResult = evaluateWinner(userMove, cpuMove)
-              println(s"Math result  : ${matchResult}\n\n")
-
-              Some(PlayAgainCommand)
-            }
-            case _ => None
-          }
+    val userInput = scala.io.StdIn.readLine()
+
+    val maybePlayAgain = Either
+      .cond(
+        !isExitCommandDetect(userInput),
+        parsePlayerMove(userInput),
+        Exit
       )
+      .map(handleUserMove)
 
-    maybePlayAgain.map(userCmd => play())
+    maybePlayAgain.fold(
+      exit => println("--- EXIT ---"),
+      playAgain => play()
```

Comment:
```suggestion
      _ => play()
```

(visto che non lo usi)